src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -7379,5 +7379,20 @@ internal string GetDebuggerDisplay()
                 return pooledBuilder.ToStringAndFree();
             }
         }
+
+        internal sealed class NullabilityInfoTypeComparer : IEqualityComparer<(NullabilityInfo info, TypeSymbol type)>
+        {
+            public bool Equals((NullabilityInfo info, TypeSymbol type) x, (NullabilityInfo info, TypeSymbol type) y)
+            {
+                return x.info.Equals(y.info) &&
+                       x.type is object &&
+                       x.type.Equals(y.type, SymbolEqualityComparer.ConsiderEverything);
```

Comment:
Nit: could just do `SymbolEqualityComparer.Equals(x.type, y.type)`, especially since this won't work for `null`.